internal/template-bundle/bundle.go

```diff
@@ -176,3 +191,35 @@ func createDataSource(name, namespace string) cdiv1beta1.DataSource {
 		},
 	}
 }
+
+func manageSingleReplicaInfrastructure(templates []templatev1.Template, topologyMode *osconfv1.TopologyMode, codec runtime.Encoder) error {
+	if *topologyMode != osconfv1.SingleReplicaTopologyMode {
+		return nil
+	}
+
+	for templateIdx, template := range templates {
+		for objIdx, rawObj := range template.Objects {
```

Comment:
These loops unnecessarily copy the objects inside the slices. Can you change them to only use indexes?
```go
for templateIdx := range templates {
	objects := templates[templateIdx].Objects
	for objIdx := range objects {
		rawObj := &objects[objIdx]
```